vnext/Microsoft.ReactNative/JsiReader.h

```diff
@@ -8,6 +11,15 @@
 
 namespace winrt::Microsoft::ReactNative {
 
+// workaround: in macOS 10.13, optional<T>::value is not available
+
+#ifndef __APPLE__
+template <typename T>
+decltype(auto) ReadOptional(std::optional<T> &opt) {
```

Comment:
>decltype(auto) [](start = 0, length = 14)

It could be just 'auto'

Reply:
Here I want to keep the same exact reference type, so I am intended to use `decltype(auto)`